pyEpiabm/pyEpiabm/sweep/initial_infected_sweep.py

```diff
@@ -51,15 +52,35 @@ def __call__(self, sim_params: dict):
             raise ValueError('There are not enough susceptible people in the \
                                         population to infect')
 
-        if ("initial_infected_cell" not in sim_params
-                or not sim_params["initial_infected_cell"]):
-            all_persons = [pers for cell in self._population.cells for pers
-                           in cell.persons if pers.infection_status
-                           == InfectionStatus.Susceptible]
+        care_param = Parameters.instance().carehome_params
+        carehome_infection = care_param["carehome_allow_initial_infections"]
+
+        if carehome_infection > 0:
```

Comment:
I have to admit I'm not a fan of the code duplication here. It's not obvious to me why we need a separate case just to exclude people associated with carehomes?  Can they not be initial seeds for infection>

Or if we want to exclude transmission in carehomes, and places have an infectiousness  parameter, maybe we just set the infectiousness of carehomes to zero?